src/client/QXmppVCardManager.cpp

```diff
@@ -113,14 +113,29 @@ bool QXmppVCardManager::handleStanza(const QDomElement& element)
         QXmppVCardIq vCardIq;
         vCardIq.parse(element);
 
-        if (vCardIq.from().isEmpty()) {
+        if (vCardIq.type() == QXmppIq::Error && vCardIq.error().type() == QXmppIq::Error::Cancel) {
+            switch(vCardIq.error().condition()) {
+            case QXmppIq::Error::ServiceUnavailable:
+                emit noVCardFound();
+                return true;
+                break;
+            case QXmppIq::Error::ItemNotFound:
+                emit noClientVCardFound();
+                return true;
+                break;
+            default:
+                return false;
+            }
+        }
+
+        if (vCardIq.from().isEmpty() || vCardIq.from() == client()->configuration().jidBare()) {
             d->clientVCard = vCardIq;
             d->isClientVCardReceived = true;
             emit clientVCardReceived();
+            return true;
```

Comment:
This breaks compatibility.

Reply:
It changes the behaviour of the vCardReceived() signal. Clients depending on that the signal is always emitted wouldn't work after this anymore.